app_gui.py

```diff
@@ -0,0 +1,159 @@
+import webbrowser
+from modules.PersonKeyManager import PersonKeyManager
+from modules.EncryptionClient import EncryptionClient
+from appdirs import AppDirs
+import sys
+import zerorpc
+import os
+from modules.VaultClient import VaultClient
+from util.config_loader import config
+import logging
+from util.util import Util
+import uuid
+from modules.DatasetKeyManager import DatasetKeyManager
+from modules.FRDRAPIClient import FRDRAPIClient
+
+
+__version__ = config.VERSION
+dirs = AppDirs(config.APP_NAME, config.APP_AUTHOR)
+os.makedirs(dirs.user_data_dir, exist_ok=True)
+
+
+class EncryptionClientGui(object):
+    def __init__(self):
+        Util.get_logger("frdr-encryption-client",
+                        log_level="info",
+                        filepath=os.path.join(dirs.user_data_dir, "frdr-encryption-client_log.txt"))
+        self._logger = logging.getLogger("frdr-encryption-client.gui")
+        self._vault_client = VaultClient()
+        self._frdr_api_client = FRDRAPIClient()
+        # self._vault_client_pki = VaultClient()
+
+    def login_oidc_google(self, hostname):
+        try:
+            self._logger.info(
+                "Log into Vault using oidc method with google acccount")
+            self._vault_client.login(vault_addr=hostname,
+                                     auth_method="oidc",
+                                     oauth_type="google")
+            return (True, None)
+        except Exception as e:
+            self._logger.info(str(e))
+            return (False, str(e))
+
+    def login_oidc_globus(self, hostname, hostname_pki):
+        try:
+            self._logger.info(
+                "Log into Vault using oidc method with globus acccount")
+            self._vault_client.login(vault_addr=hostname,
+                                     auth_method="oidc")
+            # self._vault_client_pki.login(vault_addr=hostname_pki,
+            #                          auth_method="oidc")
+            return (True, None)
+        except Exception as e:
+            self._logger.info(str(e))
+            return (False, str(e))
+
+    def login_frdr_api_globus(self, base_url=None):
+        try:
+            self._logger.info(
+                "Login with globus acccount for FRDR REST API usage")
+            if base_url is None:
+                base_url = config.FRDR_API_BASE_URL
+            self._frdr_api_client.login(base_url=base_url)
+            return (True, None)
+        except Exception as e:
+            self._logger.info(str(e))
+            return (False, str(e))
+
+    def logout(self):
+        try:
+            self._vault_client.logout()
+            webbrowser.open("https://auth.globus.org/v2/web/logout")
```

Comment:
This would also be a good candidate for a static declaration, or configuration somewhere

Reply:
Fixed. The url has been added to the config file.